sycl/source/detail/queue_impl.hpp

```diff
@@ -109,11 +105,6 @@ class queue_impl {
                             "Queue cannot be constructed with both of "
                             "discard_events and enable_profiling.");
     }
-    if (!Context->hasDevice(Device))
```

Comment:
I think we still have to check that the Device **or it's root-device** are in the context